src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -6124,3 +6123,32 @@ class CreateUserStatementSegment(ansi.CreateUserStatementSegment):
             optional=True,
         ),
     )
+
+
+class ComputedColumnDefinitionSegment(BaseSegment):
+    """A computed column definition, e.g. for CREATE TABLE or ALTER TABLE.
+
+    https://learn.microsoft.com/en-us/sql/relational-databases/tables/specify-computed-columns-in-a-table?view=sql-server-ver16
+    """
+
+    type = "computed_column_definition"
+
+    match_grammar: Matchable = Sequence(
+        Ref("SingleIdentifierGrammar"),  # Column name
```

Comment:
Should this be a ColumnReferenceSegment?

Reply:
Hi @WittierDinosaur, 
I can't tell, I'm no expert.
From the name I would argue it is something different than a _reference_ to a column. It is rather a _definition_ of column; a _rule_ how to determine a value of computed column.

Reply:
Ah yeah, fair point!